client/web-sveltekit/src/lib/repo/open-in-editor/OpenInEditor.svelte

```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+    import {getEditor, parseBrowserRepoURL, buildRepoBaseNameAndPath, buildEditorUrl} from '$lib/web'
+    import {getEditorSettingsErrorMessage} from './build-url'
+    import Tooltip from '$lib/Tooltip.svelte'
+    import EditorIcon from '$lib/repo/open-in-editor/EditorIcon.svelte'
+    import {settings} from '$lib/stores'
+    import {page} from '$app/stores';
+    import type {ExternalRepository} from '$lib/graphql-types';
+    import DefaultEditorIcon from '$lib/repo/open-in-editor/DefaultEditorIcon.svelte';
+
+    export let externalServiceType: ExternalRepository['serviceType'] = ''
+
+    let openInEditor = $settings?.openInEditor
```

Comment:
This should also be reactive since `settings` is a store and `$settings` might change over time. I don't believe it does right now but it would be a subtle bug otherwise.

```suggestion
    $: openInEditor = $settings?.openInEditor
```